### PR TITLE
Part3

### PR DESCRIPTION
**Pull request by daviddassau**

### CODE REVIEW

**Comment by daviddassau** [comment]
Firebase link: https://angular-contacts-db03c.firebaseapp.com/#!/login

**Comment by rousell** [comment]
H i @daviddassau, when I have one contact and I try to delete it, it's not deleting that contact from the UI. I'm working on getting more details for you on how to fix it

File: javascripts/services/ContactService.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    const getUserContact = (userUid) => {
+        let contact = [];
+        return $q((resolve, reject) => {
+            $http.get(`${FIREBASE_CONFIG.databaseURL}/contacts.json?orderBy="uid"&equalTo="${userUid}"`).then((results) => {
+                let fbContacts = results.data;
+                Object.keys(fbContacts).forEach((key) => {
+                    fbContacts[key].id = key;
+                    contact.push(fbContacts[key]);
+                    console.log(contact);
+                    resolve(contact);
```
Comment (rousell) [inline javascripts/services/ContactService.js]: this resolution for your contact should be outside of your looping over of your Object.keys

Comment (daviddassau) [reply javascripts/services/ContactService.js]: @rousell I believe it's fixed now!

**Comment by daviddassau** [comment]
@rousell Did you see that i was ready for you to check it again?

**Comment by rousell** [comment]
@daviddassau great, this looks good! I think the only part of part3 that you are missing now is the validation on your form: `A user should not be able to submit a contact without putting information in each input on the form (check out ng-validate to help with this)`

I'd suggest looking at this as a reference for how to implement the validation: https://scotch.io/tutorials/angularjs-form-validation

**Comment by daviddassau** [comment]
Ok @rousell Everything should be GTG now!

File: javascripts/controllers/NewCtrl.js [hunk 2]
```diff
@@ -1,5 +1,16 @@
 "use strict";
 
-app.controller("NewCtrl", function ($scope) {
-    $scope.controller = "NewCtrl";
+app.controller("NewCtrl", function ($location, $rootScope, $scope, ContactService) {
+
+    $scope.submitForm = () => {
```
Comment (zoeames) [inline javascripts/controllers/NewCtrl.js]: should this line have an input?

File: javascripts/controllers/NewCtrl.js [hunk 3]
```diff
@@ -1,5 +1,16 @@
...
+        let newContact = $scope.contact;
+        $scope.contact.uid = $rootScope.uid;
+        console.log(newContact);
```
Comment (zoeames) [inline javascripts/controllers/NewCtrl.js]: remove console.log

File: javascripts/controllers/NewCtrl.js [hunk 4]
```diff
@@ -1,5 +1,16 @@
 "use strict";
 
-app.controller("NewCtrl", function ($scope) {
-    $scope.controller = "NewCtrl";
+app.controller("NewCtrl", function ($location, $rootScope, $scope, ContactService) {
+
+    $scope.submitForm = () => {
+        let newContact = $scope.contact;
+        $scope.contact.uid = $rootScope.uid;
```
Comment (zoeames) [inline javascripts/controllers/NewCtrl.js]: shouldn't this be newContact.uid = $rootScope.uid?

File: lib/Gruntfile.js [hunk 5]
```diff
@@ -49,20 +49,14 @@ module.exports = function (grunt) {
...
                         "lib/node_modules/angular-sanitize/angular-sanitize.js",
-                        "javascripts/app.js",
-                        "javascripts/AppConstants.js",
-                        "javascripts/AppConfig.js",
-                        "javascripts/services/AuthService.js",
-                        "javascripts/controllers/LoginCtrl.js",
-                        "javascripts/controllers/FavoritesCtrl.js",
-                        "javascripts/controllers/NewCtrl.js",
-                        "javascripts/controllers/ViewCtrl.js",
+                        "javascripts/*.js",
```
Comment (zoeames) [inline lib/Gruntfile.js:57]: combine these three line into one:

"javascripts/**/*.js"

**Comment by zoeames** [comment]
@daviddassau functionality all works but some areas of your code need some cleanup.  address my comments on your code and then push up.  DELETE THE CONSOLE.LOGS that are not in .catch

File: lib/Gruntfile.js (hunk 5 above)
Comment (daviddassau) [reply lib/Gruntfile.js:57]: I combined those on my part4 branch

File: javascripts/controllers/NewCtrl.js (hunk 3 above)
Comment (daviddassau) [reply javascripts/controllers/NewCtrl.js]: console.log removed

File: javascripts/controllers/NewCtrl.js (hunk 2 above)
Comment (daviddassau) [reply javascripts/controllers/NewCtrl.js]: I'm guessing....yes? I am unsure as to what it should be.

File: javascripts/controllers/NewCtrl.js (hunk 4 above)
Comment (daviddassau) [reply javascripts/controllers/NewCtrl.js]: GTG

File: javascripts/controllers/NewCtrl.js (hunk 2 above)
Comment (daviddassau) [reply javascripts/controllers/NewCtrl.js]: Should it be contact?

**Comment by daviddassau** [comment]
Ok, everything should be fixed now 
@rousell @zoeames

**Comment by morecallan** [comment]
👍

**Pull request merged**